justfile

```diff
@@ -249,18 +277,66 @@ uats-ingress server_model="" workers_model="" cos_model="":
     #!/usr/bin/bash
     set -euxo pipefail
 
-    goss validate --retry-timeout=900s --sleep 60s --color --max-concurrent 10
+    goss validate --retry-timeout=900s --sleep 60s --color
 
     model_suffix=$(just get-model-suffix)
 
-    juju wait-for application temporal-ui-ingress --query='name == "temporal-ui-ingress" && status == "active"'
-
     tox -e uats-ingress -- \
         --server-model="${server_model:-temporal-server-uats-${model_suffix}}" \
         --workers-model="${workers_model:-temporal-workers-uats-${model_suffix}}" \
         --cos-model="${cos_model:-cos-uats-${model_suffix}}"
 
+[private]
+check-temporal-integrated-with-prometheus:
+    #!/usr/bin/bash
+    set -euxo pipefail
+
+    model_suffix=$(just get-model-suffix)
+    prometheus_ip=$(juju status --model "cos-uats-${model_suffix}" | grep prometheus/0 | awk '{print $4}')
+
+    echo $(curl -s "http://${prometheus_ip}:9090/api/v1/label/__name__/values" | grep -q "temporal_")
```

Comment:
Are we kind of duplicating this check here and in the test file? I saw a similar test case above.

Reply:
yes, i also realized that all of our existing COS tests follow a simple pattern (make HTTP request -> get json response -> validate keys). may be we can just implement all these tests as goss (using the `http` test pattern in goss + usage of jq)

im leaning towards keeping things a bit duplicated for this PR, but quickly determine if we expect to do more complex tests for COS. i will follow up with an issue in this repo to clean this up if we decide to just use goss instead of pytest